Lib/_pydecimal.py

```diff
@@ -2180,17 +2231,19 @@ def _power_exact(self, other, p):
 
         # if m > p*100//_log10_lb(xc) then m > p/log10(xc), hence xc**m >
         # 10**p and the result is not representable.
+        assert xc != 1, self
+        assert not _is_power_of_10(str(xc)), self
```

Comment:
You can use more strong and cheaper assertion:
```suggestion
        assert xc % 10 != 0, self
```